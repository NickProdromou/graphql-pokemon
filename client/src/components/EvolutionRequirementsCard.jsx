import React from "react";
import PropTypes from "prop-types";
import Styled from "styled-components";

import { type } from "../style/mixins/index";
import { spacing, colours } from "../style/variables";

const EvolutionReqirementsCard = ({ name, amount }) => (
  <StyledRequirements>
    <span className="RequirementName">
      <span className="RequirementAmount">{amount}</span>
      {name}
    </span>
  </StyledRequirements>
);

EvolutionReqirementsCard.propTypes = {
  name: PropTypes.string.isRequired,
  amount: PropTypes.number.isRequired
};

const StyledRequirements = Styled.div`
  padding: ${spacing.small.level5} ${spacing.small.level2};
  background: ${colours.borderColor};

  :not(:last-child) {
    margin-bottom: ${spacing.small.level4};
  }

  .RequirementName {
    ${type("ui")}
    margin-bottom: ${spacing.small.level3};
    margin-top: 0;
  }

  .RequirementAmount {
    ${type("ui")}
    margin-right: ${spacing.small.level1};
    font-weight: bold;
    margin-bottom: ${spacing.small.level2};
  }
`;

export default EvolutionReqirementsCard;
