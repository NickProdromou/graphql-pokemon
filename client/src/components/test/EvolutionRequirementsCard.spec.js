import React from "react";
import renderer from "react-test-renderer";
import EvolutionRequirementsCard from "../EvolutionRequirementsCard";

describe("EvolutionRequirementsCard component", () => {
  const props = {
    type: "Fire"
  };

  test("renders", () => {
    const props = {
      name: "foo",      
      amount: 23,
    };

    const component = renderer.create(
      <EvolutionRequirementsCard
        name={props.name}
        amount={props.amount}
      />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
