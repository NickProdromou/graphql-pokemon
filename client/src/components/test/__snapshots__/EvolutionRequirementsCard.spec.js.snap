// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EvolutionRequirementsCard component renders 1`] = `
<div
  className="sc-bdVaJa jUUOdF"
>
  <span
    className="RequirementName"
  >
    <span
      className="RequirementAmount"
    >
      23
    </span>
    foo
  </span>
</div>
`;
